### PR TITLE
Expand and consolidate networking docs

### LINKED ISSUE #67956: Listening addresses configuration (network.host) not specified (multiple values allowed)
Following ticket #13592 and a change my @rmuir , it has been possible for a long time to give _network.host_ several values. For example:
`network.host: [ '_local_' , '_site:ipv4_' ]`

Unfortunately, _network.host_ is unspecified and that change was accompanied by hardly any documentation update. It would be most helpful to specify, or at least add at least 1 example with multiple values.

The default configuration file pretty much implies that multiples values are unsupported:
> Set the bind address to a specific IP (IPv4 or IPv6):

### Note
[The current section](https://www.elastic.co/guide/en/elasticsearch/reference/master/important-settings.html#network.host)'s phrasing suggests that this setting is only useful to form a cluster.

### PR DESCRIPTION
**Pull request by DaveCTurner**
Today's network config docs are split into "Network", "HTTP" and
"Transport" pages, with unclear relationships between them. We often
encounter users with weird configs that indicate they don't really
understand how these settings all relate. In fact these pages are all
very interrelated, and the HTTP and Transport pages are almost all only
for advanced users. This commit brings these docs into a single page and
rewords some things to try and guide users away from the advanced
settings unless their configuration needs all the extra complexity.

It also adds a section entitled "Binding and publishing" which clarifies
the meanings of the `bind_host` and `publish_host` parameters. This is
also a common source of confusion amongst users.

It also clarifies that many of these settings accept a list of
addresses, and warns that this may not be what you want. Closes #67956.

Preview: https://elasticsearch_68051.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/modules-network.html

### CODE REVIEW

**Comment by lockewritesdocs** [comment]
@elasticmachine test this please

File: docs/reference/modules/network/tracers.asciidoc [hunk 1]
```diff
@@ -0,0 +1,72 @@
+==== Request tracing
+
+It is possible to trace individual requests made on the HTTP and transport
+layers. Be warned that this tracing can generate extremely high log volumes
+which may destabilise your cluster. Do not enable request tracing on busy or
+important clusters.
```
[the remark this answers is not in the document]
Comment (lockewritesdocs) [reply docs/reference/modules/network/tracers.asciidoc:6]: GitHub won't handle that change as a suggestion that can be easily committed. Given the gravity of the message, I'd consider making it a WARNING admonition.

File: docs/reference/modules/network/tracers.asciidoc [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+[[http-rest-request-tracer]]
+===== REST request tracer
+
+The HTTP layer has a dedicated tracer logger which, when activated, logs
+incoming requests and the corresponding outgoing responses. The log can be
+dynamically activated by setting the level of the
+`org.elasticsearch.http.HttpTracer` logger to `TRACE`:
```
Comment (lockewritesdocs) [inline docs/reference/modules/network/tracers.asciidoc:14]:
```suggestion
The HTTP layer has a dedicated tracer that logs
incoming requests and the corresponding outgoing responses. Activate
the tracer by setting the level of the
`org.elasticsearch.http.HttpTracer` logger to `TRACE`:
```

File: docs/reference/modules/network/tracers.asciidoc [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+}
+--------------------------------------------------
+
+[[transport-tracer]]
+===== Transport tracer
+
+The transport layer has a dedicated tracer logger which, when activated, logs
+incoming and outgoing requests and responses. The log can be dynamically
+activated by setting the level of the
+`org.elasticsearch.transport.TransportService.tracer` logger to `TRACE`:
```
Comment (lockewritesdocs) [inline docs/reference/modules/network/tracers.asciidoc:46]:
```suggestion
The transport layer has a dedicated tracer that logs
incoming and outgoing requests and responses. Activate the tracer 
by setting the level of the
`org.elasticsearch.transport.TransportService.tracer` logger to `TRACE`:
```

File: docs/reference/modules/transport.asciidoc [hunk 4]
```diff
@@ -1,45 +1,55 @@
...
 [[transport-settings]]
 ==== Transport settings
 
-The following settings can be configured for the internal transport that
-communicates over TCP. These settings also use the common
-<<modules-network,network settings>>.
+The following settings allow you to configure the transport interface
+independently of the <<http-settings,HTTP interface>>. You can also configure
+both interfaces together using the <<common-network-settings,network
+settings>>.
```
Comment (lockewritesdocs) [inline docs/reference/modules/transport.asciidoc:7]:
```suggestion
Use the following settings to configure the transport interface
independently of the <<http-settings,HTTP interface>>. Use the
<<common-network-settings,network
settings>> to configure both interfaces together.
```

File: docs/reference/modules/transport.asciidoc [hunk 5]
```diff
@@ -1,45 +1,55 @@
...
 
 `transport.port`::
 (<<static-cluster-setting,Static>>)
-A bind port range. Defaults to `9300-9400`.
+The port to bind for communication between nodes. Accepts a single value or a
+range. If a range is specified, the node will bind to the first available port
+in the range. You should set this setting to a single port, not a range, on
+every master-eligible node.
```
Comment (lockewritesdocs) [inline docs/reference/modules/transport.asciidoc:14]:
```suggestion
in the range. Configure this setting to use a single port only on
every master-eligible node.
```

File: docs/reference/modules/transport.asciidoc [hunk 6]
```diff
@@ -1,45 +1,55 @@
...
+transport and HTTP interfaces.
++
+Defaults to the address given by `network.host`.
 
 `transport.bind_host`::
 (<<static-cluster-setting,Static>>)
-The host address to bind the transport service to. Defaults to
-`transport.host` (if set) or `network.bind_host`.
+The network address(es) to which the node should bind in order to listen for
+incoming transport connections. Accepts a list of IP addresses, hostnames, and
```
Comment (lockewritesdocs) [inline docs/reference/modules/transport.asciidoc:31]:
```suggestion
Bind addresses the node should use to listen for incoming
transport connections. Accepts a list of IP addresses, hostnames, and
```

File: docs/reference/modules/network.asciidoc [hunk 7]
```diff
@@ -97,39 +114,86 @@ installed.
...
-This specifies which network interface(s) a node should bind to in order to
-listen for incoming requests.  A node can bind to multiple interfaces, e.g.
-two network cards, or a site-local address and a local address. Defaults to
-`network.host`.
+(<<static-cluster-setting,Static>>)
+The network address(es) to which the node should bind in order to listen for
+incoming connections. Accepts a list of IP addresses, hostnames, and
+<<network-interface-values,special values>>. Defaults to the address given by
+`network.host`. You should only use this setting if you wish to bind to
+multiple addresses or to use different addresses for publishing and binding.
```
Comment (lockewritesdocs) [inline docs/reference/modules/network.asciidoc:174]:
```suggestion
`network.host`. Use this setting only if binding to
multiple addresses or to use different addresses for publishing and binding.
```

File: docs/reference/modules/transport.asciidoc [hunk 8]
```diff
@@ -1,45 +1,55 @@
...
+different publishing configurations for the transport and HTTP interfaces.
 
-`transport.host`::
+`transport.publish_port`::
 (<<static-cluster-setting,Static>>)
-Used to set the `transport.bind_host` and the `transport.publish_host`.
+The port of the <<modules-network-binding-publishing,transport publish
+address>>. You should only set this if you need the publish port to be
+different from `transport.port`. Defaults to the port assigned via
+`transport.port`.
```
Comment (lockewritesdocs) [inline docs/reference/modules/transport.asciidoc:52]:
```suggestion
address>>. Set this parameter only if you need the publish port to be
different from `transport.port`. Defaults to the port assigned via
`transport.port`.
```

File: docs/reference/modules/transport.asciidoc [hunk 9]
```diff
@@ -1,45 +1,55 @@
...
 `transport.publish_host`::
 (<<static-cluster-setting,Static>>)
-The host address to publish for nodes in the cluster to connect to.
-Defaults to `transport.host` (if set) or `network.publish_host`.
+The network address at which the node can be contacted by other nodes. Accepts
+an IP address, a hostname, or a <<network-interface-values,special value>>.
+Defaults to the address given by `transport.host` or `network.publish_host`.
+You should only use this setting if you wish to bind to multiple addresses or
+to use different addresses for publishing and binding, and you require
+different publishing configurations for the transport and HTTP interfaces.
```
Comment (lockewritesdocs) [inline docs/reference/modules/transport.asciidoc:45]:
```suggestion
Use this setting only if you require
different binding configurations for the transport or HTTP interfaces, 
such as binding to multiple addresses or using different addresses for publishing
and binding.
```

File: docs/reference/setup/bootstrap-checks.asciidoc [hunk 10]
```diff
@@ -24,19 +24,19 @@ documented individually.
...
-By default, Elasticsearch binds to loopback addresses for <<modules-http,HTTP>>
-and <<modules-transport,transport (internal)>> communication. This is fine for
-downloading and playing with Elasticsearch as well as everyday development, but
-it's useless for production systems. To join a cluster, an Elasticsearch node
-must be reachable via transport communication. To join a cluster via a
+By default, Elasticsearch binds to loopback addresses for
+<<modules-network,HTTP and transport (internal) communication>>. This is fine
+for downloading and playing with Elasticsearch as well as everyday development,
+but it's useless for production systems. To join a cluster, an Elasticsearch
+node must be reachable via transport communication. To join a cluster via a
```
Comment (lockewritesdocs) [inline docs/reference/setup/bootstrap-checks.asciidoc:31]:
```suggestion
By default, {es} binds to loopback addresses for
<<modules-network,HTTP and transport (internal) communication>>. This is fine
for downloading and playing with {es} as well as everyday development,
but it's useless for production systems. To join a cluster, an {es}
node must be reachable via transport communication. To join a cluster via a
```

**Review by lockewritesdocs — APPROVED** [review]
This is a great improvement @DaveCTurner! I left a bunch of suggested edits that you should be able to easily commit if you decide.

**Comment by DaveCTurner** [comment]
Thanks @lockewritesdocs for all the valuable edits.

**Pull request merged**